pydevd.py

```diff
@@ -1077,7 +1077,18 @@ def get_file_type(self, frame, abs_real_path_and_basename=None, _cache_file_type
             if abs_real_path_and_basename[0] == "<string>":
                 # Consider it an untraceable file unless there's no back frame (ignoring
                 # internal files and runpy.py).
+                if frame.f_back is None:
+                    _cache_file_type[cache_key] = None
+                    return None
+                
+                back_basename = pydevd_file_utils.basename(frame.f_back.f_code.co_filename)
```

Comment:
get_file_type was failing in sys.monitoring because a bunch of <string> frames were coming from the code in pydevd.

Reply:
I definitely ran the build tools (it's how I updated the cython files).

I should remove the logging though. Forgot about that. I need to debug this again to see what the exact problem was.

Reply:
Sorry didn't get to this one yet. Need to debug still.

Reply:
Ok... this really needs some debugging (as I don't understand why `get_file_type` would return something that's not `PYDEV_FILE` for something in `sys_monitoring` -- if something is needed, the fix should probably be there as this would affect other places too, but in theory this shouldn't be needed, possibly something is different when running in debugpy).

Reply:
Okay found the problem. In debugpy the tests call pydevd.settrace (pydevd tests don't do this). Internally this calls:

```
pydevd::_locked_settrace
pydevd::set_tracing_for_untraced_contexts
pydevd::set_trace_for_frame_and_parents line 2410
lambda
```

That lambda is causing a failure on the stop_on_entry tests for debugpy because it gets marked as the entry point.

With this change here, we skip tracing that lambda.

I'm working on adding a test case to pydevd for this.

Reply:
Without Cython enabled, a similar situation happens with pydevd_sys_monitoring.py - a lambda gets marked as having file type 'none'. I have a better change I think though that will use the get_file_type instead of hardcoding it like is done here. I'll submit that once I figure out a test that reproduces the problem.